README.md

```diff
@@ -1,526 +1,917 @@
+
+```js
+let initializeX = logged(undefined, {
+  kind: "field",
+  name: "x",
+  isStatic: false,
+  isPrivate: false,
+  defineMetadata() { /**/ }
+}) ?? (initialValue) => initialValue ;
 
-<!--
-The desugarings in this article usually omit it, but metadata is desugared as such:
 class C {
-  method() {  }
+  x = initializeX(1);
 }
-let context = { kind: "method", name: "method", isStatic: false };
-C.prototype.method = annotate(C.prototype.method, context);
-C[Symbol.metadata].prototype.methods.method = {...{a: "b"}, ...(context[Symbol.metadata] ?? {})};
--->
+```
 
-Note that, since metadata is held on the class, not on the method, the metadata is not visible to earlier decorators. Metadata on classes is added to the constructor after all class decorators have run so that they are not lost by later wrapping.
+The initializer function is called with the instance of the class as `this`, so field decorators can also be used to bootstrap registration relationships. For instance, you could register children on a parent class:
 
-### `@tracked`
+```js
+const CHILDREN = new WeakMap();
 
-The `@tracked` decorator watches a field and triggers a `render()` method when the setter is called. This pattern, or patterns like it, is common in frameworks to avoid extra bookkeeping scattered throughout the application to ask for re-rendering.
+function registerChild(parent, child) {
+  let children = CHILDREN.get(parent);
 
-Decorated fields have the semantics of getter/setter pairs around an underlying piece of private storage. The decorators can wrap these getter/setter functions. `@tracked` can wrap this getter/setter pair to implement the re-rendering behavior.
+  if (children === undefined) {
+    children = [];
+    CHILDREN.set(parent, children);
+  }
 
-```mjs
-import { tracked } from "./tracked.mjs";
+  children.push(child);
+}
 
-class Element {
-  @tracked counter = 0;
+function getChildren(parent) {
+  return CHILDREN.get(parent);
+}
 
-  increment() { this.counter++; }
+function register() {
+  return function(value) {
+    registerChild(this, value);
 
-  render() { console.log(counter); }
+    return value;
+  }
 }
 
-const e = new Element();
-e.increment();  // logs 1
-e.increment();  // logs 2
+class Child {}
+class OtherChild {}
+
+class Parent {
+  @register child1 = new Child();
+  @register child2 = new OtherChild();
+}
+
+let parent = new Parent();
+getChildren(parent); // [Child, OtherChild]
 ```
 
-When fields are decorated, the "wrapped" value is an object with two properties: `get` and `set` functions that manipulate the underlying storage. They are built to be `.call()`ed with the instance of the class as a receiver. The decorator can then return a new object of the same form. (If one of the callbacks is missing, then it is left in place unwrapped.)
+#### Classes
 
-```mjs
-// tracked.mjs
+```ts
+type ClassDecorator = (value: Function, context: {
+  kind: "class";
+  name: string;
```

Comment:
what gets passed here for an anonymous class, or one that gets an inferred name? I’d assume `undefined` is appropriate for the former (the empty string probably wouldn’t be). I haven’t tested it, but i think the latter means the name could also be a Symbol